File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:

        map = {}

        for i in arr:
            if i in map:
                map[i] = map[i]+1
            else:
                map[i] = 1

        # for i in arr:
        #     if i in map.keys():
        #         map.update({i:map.get(i)+1})
        #     else:
        #         map.setdefault(i,1)


        if(len(map.values()) != len(set(map.values()))):
            return False
        else:
            return True

File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {

        Map<Integer, Integer> map = new HashMap<Integer, Integer>();

        for (int a : arr){
            if(map.containsKey(a)){
                map.put(a, map.get(a)+1);
            }
            else{
                map.put(a,1);
            }
        }

        Set<Integer> valueSet = new HashSet<>(map.values());

        return map.size() == valueSet.size();
        
    }
}